src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs

```diff
@@ -305,10 +306,70 @@ impl Container {
 
     pub async fn stop_process(&self, container_process: &ContainerProcess) -> Result<()> {
         let mut inner = self.inner.write().await;
-        inner
-            .stop_process(container_process, true)
+
+        // ignore the stop when container status has already been stopped.
+        let status = inner.init_process.get_status().await;
+        if status == ProcessStatus::Stopped {
+            return Ok(());
+        }
+
+        match container_process.process_type {
+            ProcessType::Container => {
+                inner
```

Comment:
As both `Exec` and `Container`  `ProcessType` will do `stop_process,` I think you can just do it once, and use with_context here for different process type

Reply:
fixed